beacon-chain/rpc/eth/beacon/structs.go

```diff
@@ -299,24 +290,24 @@ type Eth1Data struct {
 }
 
 type ProposerSlashing struct {
-	SignedHeader1 SignedBeaconBlockHeader `json:"signed_header_1" validate:"required"`
-	SignedHeader2 SignedBeaconBlockHeader `json:"signed_header_2" validate:"required"`
+	SignedHeader1 *SignedBeaconBlockHeader `json:"signed_header_1" validate:"required"`
+	SignedHeader2 *SignedBeaconBlockHeader `json:"signed_header_2" validate:"required"`
 }
 
 type AttesterSlashing struct {
-	Attestation1 IndexedAttestation `json:"attestation_1" validate:"required"`
-	Attestation2 IndexedAttestation `json:"attestation_2" validate:"required"`
+	Attestation1 *IndexedAttestation `json:"attestation_1" validate:"required"`
+	Attestation2 *IndexedAttestation `json:"attestation_2" validate:"required"`
 }
 
 type Attestation struct {
-	AggregationBits string          `json:"aggregation_bits" validate:"required"`
-	Data            AttestationData `json:"data" validate:"required"`
-	Signature       string          `json:"signature" validate:"required"`
+	AggregationBits string           `json:"aggregation_bits" validate:"required"`
+	Data            *AttestationData `json:"data" validate:"required"`
+	Signature       string           `json:"signature" validate:"required"`
 }
 
 type Deposit struct {
-	Proof []string    `json:"proof" validate:"required"`
-	Data  DepositData `json:"data" validate:"required"`
+	Proof []string     `json:"proof" validate:"required,dive,hexadecimal"`
```

Comment:
is it worth adding this check for hexadecimal here? we have to convert it anyways

Reply:
Very good question. Actually our decoding error message is far superior than the one from the validate package. For now we can leave it, but we should reevaluate if we even want to use `validate` since we need to decode anyway.